src/util/io_limiter.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+use std::option::Option;
+use std::io::{Result, Write};
+
+use rocksdb::RateLimiter;
+
+const REFILL_PERIOD: i64 = 100 * 1000;
+const FARENESS: i32 = 10;
+pub const DEFAULT_SNAP_MIN_BYTES_PER_TIME: u64 = 128 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_TIME: u64 = 2 * 1024 * 1024;
+pub const DEFAULT_SNAP_MAX_BYTES_PER_SEC: u64 = 20 * 1024 * 1024;
+
+pub struct IOLimiter {
+    inner: Option<RateLimiter>,
+    min_bytes_per_time: i64,
+    max_bytes_per_time: i64,
+}
+
+impl IOLimiter {
+    pub fn new(min_bytes_per_time: u64, max_bytes_per_time: u64, bytes_per_sec: u64) -> IOLimiter {
+        let limiter = if bytes_per_sec > 0 {
+            Some(RateLimiter::new(
+                bytes_per_sec as i64,
+                REFILL_PERIOD,
+                FARENESS,
+            ))
+        } else {
+            None
+        };
+        IOLimiter {
+            inner: limiter,
+            min_bytes_per_time: min_bytes_per_time as i64,
+            max_bytes_per_time: max_bytes_per_time as i64,
+        }
+    }
+
+    pub fn set_bytes_per_second(&self, bytes_per_sec: i64) {
+        if let Some(ref limiter) = self.inner {
+            limiter.set_bytes_per_second(bytes_per_sec)
+        }
+    }
+
+    pub fn request(&self, bytes: i64) {
+        if let Some(ref limiter) = self.inner {
+            limiter.request(bytes, 1)
```

Comment:
why 1 here?

Reply:
It's a priority flag, 1 means high priority.

Reply:
use a const var